lib/modules/apostrophe-docs/lib/api.js

```diff
@@ -469,6 +475,84 @@ module.exports = function(self, options) {
     return self.deleteFromTrash(req, idOrCriteria, callback);
   };
 
+  // After moving document to the trash, prefix any fields that have
+  // unique indexes so that other pieces are allowed to reuse
+  // those usernames, email addresses, etc.
+
+  self.deduplicateTrash = function(req, doc, callback) {
+    var prefix = 'deduplicate-' + doc._id + '-';
+    var $set = {};
+    _.each(self.trashPrefixFields, function(name) {
```

Comment:
Rather than having only one array of trashPrefixFields, we need to flexibly get this based on the doc type. So we write self.getManager(doc.type).trashPrefixFields.